install/install.sh

```diff
@@ -45,328 +45,362 @@ error() {
   printf "%s\n" "${RED}x $*${NO_COLOR}" >&2
 }
 
-complete() {
+success() {
   printf "%s\n" "${GREEN}✓${NO_COLOR} $*"
 }
 
-# Gets path to a temporary file, even if
-get_tmpfile() {
-  local suffix
-  suffix="$1"
-  if hash mktemp; then
-    printf "%s.%s" "$(mktemp)" "${suffix}"
-  else
-    # No really good options here--let's pick a default + hope
-    printf "/tmp/starship.%s" "${suffix}"
-  fi
+has() {
+  command -v "$1" >/dev/null 2>&1
 }
 
-# Test if a location is writeable by trying to write to it. Windows does not let
-# you test writeability other than by writing: https://stackoverflow.com/q/1999988
-test_writeable() {
-  local path
-  path="${1:-}/test.txt"
-  if touch "${path}" 2>/dev/null; then
-    rm "${path}"
-    return 0
-  else
+mktmpdir() {
+  tmpdir=$(mktemp -dqt "$1") || {
+    error "Can't create temp file, exiting…"
     return 1
-  fi
+  }
+  echo "$tmpdir"
 }
 
 fetch() {
-  local command
-  if hash curl 2>/dev/null; then
-    set +e
-    command="curl --silent --fail --location $1"
-    curl --silent --fail --location "$1"
-    rc=$?
-    set -e
-  else
-    if hash wget 2>/dev/null; then
-      set +e
-      command="wget -O- -q $1"
-      wget -O- -q "$1"
-      rc=$?
-      set -e
+  file="$1"
+  url="$2"
+
+  if has curl; then
+    cmd="curl -w '%{http_code}' -sL -o $file $url"
+    code=$(curl -w '%{http_code}' -sL -o "$file" "$url")
+    if [ "$code" = "200" ]; then
```

Comment:
Unlike my previous comment on mktempdir, this can easily be simulated in production by just `curl`ing a large file and then shutting off your network. The HTTP code will be 200, but the file will be incomplete.

Reply:
Wow, that's a really good catch! I went with HTTP codes because the curl won't fail on 404s.

Reply:
Does the `--fail` flag make it fail on 404s?

According to the link, it doesn't work for some 4xx errors involving auth, but I don't think we need to worry about that.

https://superuser.com/questions/590099/can-i-make-curl-fail-with-an-exitcode-different-than-0-if-the-http-status-code-i

Reply:
Actually if used with `--fail` it will result in 22. Ok, time to revert this. 😉